sormas-backend/src/main/java/de/symeda/sormas/backend/share/ExternalShareInfoFacadeEjb.java

```diff
@@ -78,6 +83,26 @@ public List<ExternalShareInfoDto> getIndexList(ExternalShareInfoCriteria criteri
 		return shareInfoList.stream().map(i -> convertToDto(i, pseudonymizer)).collect(Collectors.toList());
 	}
 
+	@Override
+	public boolean isSharedCase(String caseUuid) {
+		final CriteriaBuilder cb = em.getCriteriaBuilder();
+		final CriteriaQuery<ExternalShareStatus> cq = cb.createQuery(ExternalShareStatus.class);
```

Comment:
I've also tested this PR and I've found some problems related to the filtering.

I have the below 4 cases ( the first 3 are shared, the last one is not shared with the Reporting Tool)
![AllTheCases](https://user-images.githubusercontent.com/47103965/202163219-c5bca952-92c8-4a9c-8e07-9bff865234e2.PNG)
![AllTheCases2](https://user-images.githubusercontent.com/47103965/202163239-2f974f08-6871-4054-8111-a6949470f56b.PNG)

1) When I filter "Only cases already shared with Reporting Tool" one case will not be retrieved : S35GZR.
![S35GZRMissingAsSharedBecauseTheFlagIsTrue](https://user-images.githubusercontent.com/47103965/202163686-922f4a7d-379e-443d-a323-6be39b24fb37.PNG)
This case was shared with the Reporting Tool but after sharing the user clicked the "Don't share..." flag which for future cases will not be a problem but for existent , old cases should work properly.
![image](https://user-images.githubusercontent.com/47103965/202164063-9fb27d2f-ff93-4dff-ae7d-99949359614f.png)


2)For filtering with "Only cases not yet shared with reporting tool" 1 case(WCMCE2) should be retrieved and 0 case is retrieved.
![OutcomeOfTheFiltering](https://user-images.githubusercontent.com/47103965/202164379-216a60c4-1066-447c-b1b9-a2a49612afc1.PNG)

The case has the "Don't share with..." flag but never was shared with Reporting Tool so the case should be retrieved for this type of filtering.

![image](https://user-images.githubusercontent.com/47103965/202164549-9294c9eb-e6a2-410f-8aae-787b99011944.png)


3) When I click the filtering "Only cases changed since last shared with ..." I receive an error.
![ONly cases changed since](https://user-images.githubusercontent.com/47103965/202164927-db68b6e5-4666-4048-b838-823a35a17834.PNG)

4) The last type of filtering related to the sharing with reporting tool, "Only cases marked with Don't share with reporting tool" is working fine.